nvme.c

```diff
@@ -5317,6 +5321,330 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
+	const char *pmrcap = "PMRCAP=0xe00 register offset";
+	const char *pmrctl = "PMRCTL=0xe04 register offset";
+	const char *pmrsts = "PMRSTS=0xe08 register offset";
+	const char *pmrebs = "PMREBS=0xe0c register offset";
+	const char *pmrswtp = "PMRSWTP=0xe10 register offset";
+	const char *pmrmscl = "PMRMSCL=0xe14 register offset";
+	const char *pmrmscu = "PMRMSCU=0xe18 register offset";
+
+	_cleanup_nvme_dev_ struct nvme_dev *dev = NULL;
+	int err;
+	enum nvme_print_flags flags;
+	bool fabrics = false;
+	_cleanup_nvme_root_ nvme_root_t r = NULL;
+	void *bar;
+
+	struct config {
+		int offset;
+		bool human_readable;
+		bool cap;
+		bool vs;
+		bool intms;
+		bool intmc;
+		bool cc;
+		bool csts;
+		bool nssr;
+		bool aqa;
+		bool asq;
+		bool acq;
+		bool cmbloc;
+		bool cmbsz;
+		bool bpinfo;
+		bool bprsel;
+		bool bpmbl;
+		bool cmbmsc;
+		bool cmbsts;
+		bool crto;
+		bool pmrcap;
+		bool pmrctl;
+		bool pmrsts;
+		bool pmrebs;
+		bool pmrswtp;
+		bool pmrmscl;
+		bool pmrmscu;
+	};
+
+	struct config cfg = {
+		.offset		= -1,
+	};
+
+	NVME_ARGS(opts, cfg,
+		 OPT_UINT("offset",         'O', &cfg.offset,         offset),
+		 OPT_FLAG("human-readable", 'H', &cfg.human_readable, human_readable),
+		 OPT_FLAG("cap",            'c', &cfg.cap,            cap),
+		 OPT_FLAG("vs",             'V', &cfg.vs,             vs),
+		 OPT_FLAG("intms",          'i', &cfg.intms,          intms),
+		 OPT_FLAG("intmc",          'I', &cfg.intmc,          intmc),
+		 OPT_FLAG("cc",             'c', &cfg.cc,             cc),
+		 OPT_FLAG("csts",           'C', &cfg.csts,           csts),
+		 OPT_FLAG("nssr",           'n', &cfg.nssr,           nssr),
+		 OPT_FLAG("aqa",            'a', &cfg.aqa,            aqa),
+		 OPT_FLAG("asq",            'A', &cfg.asq,            asq),
+		 OPT_FLAG("acq",            'q', &cfg.acq,            acq),
+		 OPT_FLAG("cmbloc",         'm', &cfg.cmbloc,         cmbloc),
+		 OPT_FLAG("cmbsz",          'M', &cfg.cmbsz,          cmbsz),
+		 OPT_FLAG("bpinfo",         'b', &cfg.bpinfo,         bpinfo),
+		 OPT_FLAG("bprsel",         'B', &cfg.bprsel,         bprsel),
+		 OPT_FLAG("bpmbl",          'p', &cfg.bpmbl,          bpmbl),
+		 OPT_FLAG("cmbmsc",         's', &cfg.cmbmsc,         cmbmsc),
+		 OPT_FLAG("cmbsts",         'S', &cfg.cmbsts,         cmbsts),
+		 OPT_FLAG("crto",           'r', &cfg.crto,           crto),
+		 OPT_FLAG("pmrcap",         'P', &cfg.pmrcap,         pmrcap),
+		 OPT_FLAG("pmrctl",         'R', &cfg.pmrctl,         pmrctl),
+		 OPT_FLAG("pmrsts",         't', &cfg.pmrsts,         pmrsts),
+		 OPT_FLAG("pmrebs",         'e', &cfg.pmrebs,         pmrebs),
+		 OPT_FLAG("pmrswtp",        'w', &cfg.pmrswtp,        pmrswtp),
+		 OPT_FLAG("pmrmscl",        'l', &cfg.pmrmscl,        pmrmscl),
+		 OPT_FLAG("pmrmscu",        'u', &cfg.pmrmscu,        pmrmscu));
+
+	err = parse_and_open(&dev, argc, argv, desc, opts);
+	if (err)
+		return err;
+
+	if (!argconfig_parse_seen(opts, "offset")) {
```

Comment:
I think I suggested this last time, but can this not also be changed to `cfg.offset < 0`?

Reply:
Sorry missed to understand the suggenstion so fix by the commit 301da031 as mentioned.